libdnf/base/goal.cpp

```diff
@@ -406,13 +405,13 @@ void Goal::Impl::add_upgrades_to_goal() {
             // TODO(jmracek) Replace obsoletes_query.get_package_set(); by more effective approach
             // TODO(jmracek) use upgrades + installed when the filter will be available libdnf::rpm::SolvQuery what_obsoletes(query);
             // what_obsoletes.ifilter_upgrades()
-            obsoletes_query.ifilter_obsoletes(libdnf::sack::QueryCmp::EQ, query.get_package_set());
+            obsoletes_query.ifilter_obsoletes(libdnf::sack::QueryCmp::EQ, query);
             // obsoletes = self.sack.query().available().filterm(obsoletes=installed_query.union(q.upgrades()))
-            query.p_impl->query_result |= obsoletes_query.p_impl->query_result;
+            *query.pImpl |= *obsoletes_query.pImpl;
```

Comment:
What about to make it even more simple like `query |= obsoletes_query;`?

Reply:
It was an optimization to bypass the sack test. But I'll change it. The code will be more readable.